src/serializer/ResidualHeapVisitor.js

```diff
@@ -194,7 +196,7 @@ export class ResidualHeapVisitor {
       visitedBindings[n] = visitedBinding;
     }
     invariant(visitedBinding.value !== undefined);
-    this.visitValue(visitedBinding.value);
+    visitedBinding.value = this.visitEquivalentValue(visitedBinding.value);
```

Comment:
But I think there's a good chunk remaining visitValue calls where non-object values appear. For example in visitValueMap the keys and values can be any Value.